File: src/Z.Collections/System.Collections.Specialized.NameValueCollection/NameValueCollection.ToDictionary.cs
// Description: C# Extension Methods Library to enhances the .NET Framework by adding hundreds of new methods. It drastically increases developers productivity and code readability. Support C# and VB.NET
// Website & Documentation: https://github.com/zzzprojects/Z.ExtensionMethods
// Forum: https://github.com/zzzprojects/Z.ExtensionMethods/issues
// License: https://github.com/zzzprojects/Z.ExtensionMethods/blob/master/LICENSE
// More projects: http://www.zzzprojects.com/
// Copyright © ZZZ Projects Inc. 2014 - 2016. All rights reserved.
using System.Collections.Generic;
using System.Collections.Specialized;

public static partial class Extensions
{
    /// <summary>
    ///     A NameValueCollection extension method that converts the @this to a dictionary.
    /// </summary>
    /// <param name="this">The @this to act on.</param>
    /// <returns>@this as an IDictionary&lt;string,object&gt;</returns>
    public static IDictionary<string, object> ToDictionary(this NameValueCollection @this)
    {
        var dict = new Dictionary<string, object>();

        if (@this != null)
        {
            foreach (string key in @this.AllKeys)
            {
                dict.Add(key, @this[key]);
            }
        }

        return dict;
    }
}

File: src (VB.NET)/Z.ExtensionMethods.VB/Z.Collections/System.Collections.Specialized.NameValueCollection/NameValueCollection.ToDictionary.vb

' Copyright (c) 2015 ZZZ Projects. All rights reserved
' Licensed under MIT License (MIT) (https://github.com/zzzprojects/Z.ExtensionMethods)
' Website: http://www.zzzprojects.com/
' Feedback / Feature Requests / Issues : http://zzzprojects.uservoice.com/forums/283927
' All ZZZ Projects products: Entity Framework Extensions / Bulk Operations / Extension Methods /Icon Library

Imports System.Collections.Generic
Imports System.Collections.Specialized

Public Module Extensions_37

	''' <summary>
	'''     A NameValueCollection extension method that converts the @this to a dictionary.
	''' </summary>
	''' <param name="this">The @this to act on.</param>
	''' <returns>@this as an IDictionary&lt;string,object&gt;</returns>
	<System.Runtime.CompilerServices.Extension> _
	Public Function ToDictionary(this As NameValueCollection) As IDictionary(Of String, Object)
		Dim dict = New Dictionary(Of String, Object)()

		If this IsNot Nothing Then
			For Each key As String In this.AllKeys
				dict.Add(key, this(key))
			Next
		End If

		Return dict
	End Function
End Module


